tests/functional/tox_faster_test.py

```diff
@@ -1,2 +1,7 @@
+from unittest.mock import sentinel
+
+import tox_faster
+
+
 def test_it():
-    assert True
+    assert tox_faster.tox_runenvreport(sentinel.venv, sentinel.action) == []
```

Comment:
This functional tests tests a couple of things that the unit tests don't:

* You can import `tox_runenvreport` from `tox_faster` directly (rather than from `tox_faster.main`) which is what tox does
* It works with the real (non-patched) `os.environ`

The functest doesn't need to test the different possible inputs and outputs because the unit tests have that covered.

To be honest I don't really think that tox-faster needs functests but the cookiecutter generates all the functests stuff (`make functests`, `tox -e functests`, running the functests on CI, ...) and that means you need to have at least one functest because pytest fails if it collects zero tests. I could add an option to the cookiecutter to turn functests off but I don't really want to do that: I don't want some projects to have `make functests` and others not.